app/src/main/java/com/islaharper/dawnofandroid/domain/model/ApiResponse.kt

```diff
@@ -0,0 +1,17 @@
+package com.islaharper.dawnofandroid.domain.model
+
+import kotlinx.serialization.Serializable
+import kotlinx.serialization.Transient
+
+@Serializable
+data class ApiResponse(
```

Comment:
We discussed a solution in the other project using sealed classes. This current `ApiResponse` class allows for `success = true` and `error` to be non-null. We want to define invalid states like `success = true` and a non-null error out of existence. Can you please look at the previous discussion and have another go at modeling the `ApiResponse` here?